src/datasets/features.py

```diff
@@ -759,6 +807,12 @@ def encode_nested_example(schema, obj):
 def generate_from_dict(obj: Any):
     """Regenerate the nested feature object from a serialized dict.
     We use the '_type' fields to get the dataclass name to load.
+
+    generate_from_dict is the recursive helper for Features.from_dict, and allows for a convenient constructor syntax
```

Comment:
docstring I left while I was working locally - happy to remove or keep based on guidance from the maintainers!